model/src/main/kotlin/Hash.kt

```diff
@@ -40,6 +40,11 @@ data class Hash(
     val value: String
 ) {
     companion object {
+        /**
+         * A constant to specify no hash at all.
+         */
+        val NONE = Hash(HashAlgorithm.NONE, HashAlgorithm.NONE.toString())
+
         /**
          * A constant to specify an unknown hash.
```

Comment:
Is this now the constant to specify a `known` hash for an `unknown` hash algorithm?
I find it confusing that a constant for this is defined as the hash value itself can be arbitrary: is this constant used and or how about removing it?

Reply:
> Is this now the constant to specify a `known` hash for an `unknown` hash algorithm?

No, this constant would be used to say "this thing does not have a hash at all". For an existign hash value generated by an unknown hash algorithm the `UNKNOWN` constant would be continued to used.

> I find it confusing that a constant for this is defined as the hash value itself can be arbitrary: is this constant used and or how about removing it?

It cannot be arbitrary. For `NONE` the hash value is always blank (which includes it being empty). So far the constant is not used, but is it added for consistency with the `UNKNOWN` constant. I plan to use the `NONE` constant later.

Reply:
Sorry for confusion. I was too unclear, my comment was actually referring to the `UNKNOWN` constant.

Reply:
So, then yes, `UNKNOWN` is used to "specify a known hash for an unknown hash algorithm". It cannot be the other way around anyway if you think about it: If the hash value (not the algorithm) was unknown, then we would have no value, and if we have no value, we also do not know with algorithm it belongs to.

Also, in my wording I distinguish between "hash value", "hash algorithm" and "hash" (which is the combination of the former two). I'll try to make this clearer in the docs.